app/models/user/roles.rb

```diff
@@ -61,7 +63,22 @@ def provider_admin?
     admin? && account && account.provider?
   end
 
+  def non_impersonation_admin?
+    admin? && !impersonation_admin?
+  end
+
   def set_role
     self.role ||= :member
   end
+
+  private
+
+  def account_first_admin?
+    account.first_admin_id == id
+  end
+
+  def update_account_first_admin
```

Comment:
The logic should not belong to User but rather Account

Reply:
![image](https://user-images.githubusercontent.com/11318903/49794774-abf8b800-fd38-11e8-9166-37ebc9513220.png)

![image](https://user-images.githubusercontent.com/11318903/49794791-b4e98980-fd38-11e8-9e54-43c36f90cc65.png)

![image](https://user-images.githubusercontent.com/11318903/49794802-bd41c480-fd38-11e8-9005-d37be22211b9.png)

![image](https://user-images.githubusercontent.com/11318903/49794824-c7fc5980-fd38-11e8-90e1-05cb09df3382.png)

![image](https://user-images.githubusercontent.com/11318903/49794837-d0549480-fd38-11e8-8b3c-a27acf146198.png)

Reply:
![image](https://user-images.githubusercontent.com/11318903/49795032-4a851900-fd39-11e8-8cfe-ab1c35bb425e.png)